plugins/dragdropselected.lua

```diff
@@ -29,149 +60,171 @@ local function isInSelection(iLine, iCol, iSelLine1, iSelCol1, iSelLine2, iSelCo
   return true
 end -- isInSelection
 
--- distance between two points
-local function distance(x1, y1, x2, y2)
-  return math.sqrt((x2-x1)^2+(y2-y1)^2)
-end
 
-local min_drag = style.code_font:get_width(" ")
-
--- override DocView:on_mouse_moved
 local on_mouse_moved = DocView.on_mouse_moved
 function DocView:on_mouse_moved(x, y, ...)
+  if not config.plugins.dragdropselected.enabled or not self.dnd_sText then
+    -- there is nothing to do -> hand off to original on_mouse_moved()
+    return on_mouse_moved(self, x, y, ...)
+  end
 
-  local sCursor = nil
-
-  -- make sure we only act if previously on_mouse_pressed was in selection
-  if
-    self.bClickedIntoSelection
-    and
-    ( -- we are already dragging or we moved enough to start dragging
-      not self.drag_start_loc or
-      distance(self.drag_start_loc[1],self.drag_start_loc[2], x, y) > min_drag
-    )
-  then
-    self.drag_start_loc = nil
-
+  -- not sure we need to do this or if we better not
+  DocView.super.on_mouse_moved(self, x, y, ...)
+  if self.dnd_bDragging then
+    -- remove last caret showing insert location
+    self.doc:remove_selection(self.doc.last_selection)
+  else
+    self.dnd_bDragging = true
     -- show that we are dragging something
-    sCursor = 'hand'
-
-    -- calculate line and column for current mouse position
-    local iLine, iCol = self:resolve_screen_position(x, y)
-    local iSelLine1 = self.dragged_selection[1]
-    local iSelCol1  = self.dragged_selection[2]
-    local iSelLine2 = self.dragged_selection[3]
-    local iSelCol2  = self.dragged_selection[4]
-    self.doc:set_selection(iSelLine1, iSelCol1, iSelLine2, iSelCol2)
-    if not isInSelection(iLine, iCol, iSelLine1, iSelCol1, iSelLine2, iSelCol2) then
-      -- show cursor only if outside selection
-      self.doc:add_selection(iLine, iCol)
-    end
-    -- update scroll position
-    self:scroll_to_line(iLine, true)
-  end -- if previously clicked into selection
-
-  -- hand off to 'old' on_mouse_moved()
-  on_mouse_moved(self, x, y, ...)
-  -- override cursor as needed
-  if sCursor then self.cursor = sCursor end
-
+    self.cursor = 'hand'
+    -- make sure selection is marked
+    self.doc:set_selection(table.unpack(self.dnd_lSelection))
+  end
+  -- calculate line and column for current mouse position
+  local iLine, iCol = self:resolve_screen_position(x, y)
+  -- show insert location (unfortunately it doesn't always show, even when
+  -- calling draw_caret)
```

Comment:
You could set `config.disable_blink` (and restore it later).

Reply:
Thanks, I'll try that.

Reply:
works like a charm